torchvision/datasets/svhn.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import print_function
+import scipy.io as sio
+import torch.utils.data as data
+from PIL import Image
+import os
+import os.path
+import errno
+import numpy as np
+
+class SVHN(data.Dataset):
+    url = ""
+    filename = ""
+    file_md5 = ""
+    
+    def __init__(self, root, dataset='train', transform=None, target_transform=None, download=False):
+        self.root = root
+        self.transform = transform
+        self.target_transform = target_transform
+        self.dataset = dataset  # training set or test set or extra set
```

Comment:
maybe `dataset` is a misleading name, and something like `split` would be better?